[CLS-11150] feature: Retrieve all IAM Roles and its relations

Ajaya-nayak opened the pull request:
[CLS-11150] feature: Retrieve all IAM Roles and its relations in custodian service

[CLS-11150]: https://sentinelone.atlassian.net/browse/CLS-11150?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Ajaya-nayak committed:
[CLS-11150] feature: Retrieve all IAM Roles and its relations in custodian service

[CLS-11150] feature: Retrieve all IAM Roles and its relations in custodian service

vikashkumarsinhasentinelone commented on c7n/resources/iam.py:
```diff
@@ -1119,6 +1119,91 @@ def process(self, resources, event=None):
+    permissions = ('iam:ListAttachedUserPolicies',)
+
+    def _role_managed_policies(self, client, resource, value):
+        attachedPolicies = client.list_attached_role_policies(RoleName=resource['RoleName'])['AttachedPolicies']
```
Please move your role filters to the file's end.e. When we merge from the public repository of the cloud custdoan. We can simply identify our filter and merge it.

vikashkumarsinhasentinelone commented on c7n/resources/iam.py:
```diff
@@ -1119,6 +1119,91 @@ def process(self, resources, event=None):
+        inlinePolicies = client.list_role_policies(RoleName=resource['RoleName'])['PolicyNames']
+        resource['c7n:InlinePolicies'] = []
+        for inlinePolicy in inlinePolicies:
+            inlinePolicyDocument = client.get_role_policy(RoleName=resource['RoleName'], PolicyName=inlinePolicy)
```
Please fix the variable name:

https://stackoverflow.com/questions/159720/what-is-the-naming-convention-in-python-for-variable-and-function

vikashkumarsinhasentinelone commented on c7n/resources/iam.py:
```diff
@@ -1119,6 +1119,91 @@ def process(self, resources, event=None):
+class IamRoleInlinePolicy(Filter):
+    """
+        Filter IAM roles that have an inline-policy attached
+
```
Please fix the variable name:

https://stackoverflow.com/questions/159720/what-is-the-naming-convention-in-python-for-variable-and-function

Ajaya-nayak committed:
[CLS-11150] feature: Retrieve all IAM Roles and its relations in custodian service

[CLS-11150] feature: Retrieve all IAM Roles and its relations in custodian service
